lib/src/web_socket/common/web_socket.dart

```diff
@@ -229,9 +233,14 @@ abstract class CommonWebSocket extends Stream implements WebSocket {
     _error = error;
     _stackTrace = stackTrace;
 
-    // Close both incoming and outgoing communication.
-    _outgoing.close();
-    closeWebSocket(code, reason);
+    // Calling close() during a in-progress call to addStream() will result in a
+    // StateError being thrown. Avoid this by waiting for the in-progress
+    // addStream() call to complete first, if applicable.
+    (_inProgressAddStream ?? new Future(() {})).catchError((_) {}).then((_) {
+      // Close both incoming and outgoing communication.
+      _outgoing.close();
+      closeWebSocket(code ?? 1000, reason);
+    });
```

Comment:
Oh, I see now. Thanks for the comment... if only I had read further I would have had my answer.